docs/guides/open-publication.md

```diff
@@ -1,33 +1,58 @@
 # Opening a publication
 
-:warning: The described components are is still experimental.
+:warning: The APIs described here may still undergo changes before the stable 3.0 release.
 
-Readium requires you to instantiate a few components before you can actually open a publication. 
+To open a publication with Readium, you need to instantiate a couple of components: an `AssetRetriever` and a `PublicationOpener`.
 
-## Constructing an `AssetRetriever`
+## `AssetRetriever`
 
-First, you need to instantiate an `HttpClient` to provide the toolkit the ability to do HTTP requests.
-You can use the Readium `DefaultHttpClient` or a custom implementation. In the former case, its callback will
-enable you to perform authentication when required.
-Then, you can create an `AssetRetriever` which will enable you to read content through different schemes and guess its format.
-In addition to an `HttpClient`, the `AssetRetriever` constructor takes a `ContentResolver` to support data access through the `content` scheme.
+The `AssetRetriever` grants access to the content of an asset located at a given URL, such as a publication package, manifest, or LCP license.
+
+### Constructing an `AssetRetriever`
+
+You can create an instance of `AssetRetriever` with:
+
+* A `ContentResolver` to support data access through the `content` URL scheme.
+* An `HttpClient` to enable the toolkit to perform HTTP requests and support the `http` and `https` URL schemes. You can use `DefaultHttpClient` which provides callbacks for handling authentication when needed.
 
 ```kotlin
 val httpClient = DefaultHttpClient()
-
 val assetRetriever = AssetRetriever(context.contentResolver, httpClient)
 ```
 
-## Constructing a `PublicationOpener`
+### Retrieving an `Asset`
 
-The component which can parse an `Asset` giving access to a publication to build a `Publication`
-object is the `PublicationOpener`. Its constructor requires you to pass in:
+With your fresh instance of `AssetRetriever`, you can open an `Asset` from an `AbsoluteUrl`.
 
-* a `PublicationParser` it delegates the parsing work to.
-* a list of `ContentProtection`s which will deal with DRMs.
+```kotlin
+// From a `File`
+val url = File("...").toUrl()
+// or from a content:// `Uri`
+val url = contentUri.toAbsoluteUrl()
+// or from a raw URL string
+val url = AbsoluteUrl("https://domain/book.epub")
+
+val asset = assetRetriever.retrieve(url)
+    .getOrElse {  /* Failed to retrieve the Asset */  }
+```
 
-The easiest way to get a `PublicationParser` is to use the `DefaultPublicationParser` class. As to
-`ContentProtection`s, you can get one to support LCP publications through your `LcpService` if you have one.
+The `AssetRetriever` will sniff the format of the asset, which you can store in your bookshelf database to speed up the process next time you retrieve the `Asset`. This will improve performance, especially with HTTP URL schemes.
```

Comment:
Confusing. The `Format` is not what's supposed to be stored (it's not even deserializable).